pilot/helpers/agents/Architect.py

```diff
@@ -4,32 +4,34 @@
 from termcolor import colored
 from const.function_calls import ARCHITECTURE
 
-from utils.utils import execute_step, find_role_from_step, generate_app_data
+from utils.utils import should_execute_step, find_role_from_step, generate_app_data
 from database.database import save_progress, get_progress_steps
 from logger.logger import logger
 from prompts.prompts import get_additional_info_from_user
 from helpers.AgentConvo import AgentConvo
 
+ARCHITECTURE_STEP = 'architecture'
+
 
 class Architect(Agent):
     def __init__(self, project):
         super().__init__('architect', project)
         self.convo_architecture = None
 
     def get_architecture(self):
-        self.project.current_step = 'architecture'
-        self.convo_architecture = AgentConvo(self)
+        self.project.current_step = ARCHITECTURE_STEP
 
         # If this app_id already did this step, just get all data from DB and don't ask user again
-        step = get_progress_steps(self.project.args['app_id'], self.project.current_step)
-        if step and not execute_step(self.project.args['step'], self.project.current_step):
+        step = get_progress_steps(self.project.args['app_id'], ARCHITECTURE_STEP)
+        if step and not should_execute_step(self.project.args['step'], ARCHITECTURE_STEP):
             step_already_finished(self.project.args, step)
             return step['architecture']
 
         # ARCHITECTURE
         print(colored(f"Planning project architecture...\n", "green", attrs=['bold']))
         logger.info(f"Planning project architecture...")
 
+        self.convo_architecture = AgentConvo(self)
```

Comment:
why did you move setting of convo_architecture here?

In this case if we load this step from DB, in later steps convo_architecture will be undefined. I think with current implementation it might not matter but would be good to have it for future

Reply:
Because I'm used to Sonarqube complaining that variables are declared too far from where they are first used.

Following YAGNI, `convo_architecture` should probably be a local variable _until_ we do actually have a need for it to be an instance variable.